src/etl/ReportingETL.cpp

```diff
@@ -935,6 +1100,51 @@ ReportingETL::loadCache(uint32_t seq)
         assert(false);
         return;
     }
+
+    if (clioPeers.size() > 0)
+    {
+        boost::asio::spawn(
+            ioContext_, [this, seq](boost::asio::yield_context yield) {
+                for (auto& peer : clioPeers)
+                {
+                    // returns true on success
+                    if (loadCacheFromClioPeer(
+                            seq, peer.ip, std::to_string(peer.port), yield))
+                        return;
+                }
+                // if we couldn't successfully load from any peers, load from db
+                loadCacheFromDb(seq);
+            });
+        return;
+    }
+    else
+    {
+        loadCacheFromDb(seq);
+    }
+    // If loading synchronously, poll cache until full
+    while (cacheLoadStyle_ == CacheLoadStyle::SYNC &&
+           !backend_->cache().isFull())
+    {
+        BOOST_LOG_TRIVIAL(debug)
+            << "Cache not full. Cache size = " << backend_->cache().size()
+            << ". Sleeping ...";
+        std::this_thread::sleep_for(std::chrono::seconds(10));
```

Comment:
seeing that we have an `ioContext_` here, maybe it would be better to use a deadline timer so that the thread does not have to freeze? i'm not sure if we are inside an `asio`-managed context here or our own thread tho so it's just an idea - it may be good as is.

Reply:
We are in our own thread here, not an `asio` managed context, so we actually have to block. The semantics of the function are to block until the cache is loaded in this case.

Reply:
sounds good then :+1: